### PR TITLE
feat: Add Google Oauth

### PR DESCRIPTION
**Pull request by 0sinf**
## What does this PR do?

Using @nestjs/passport, Implement google oauth login.

## Type of change

- [x] New feature
- [ ] Bug fix
- [ ] Code improvements
- [ ] Breaking change
- [ ] Documentation
- [ ] etc.

## Changes

- Add @nestjs/passport, passport, passport-google-oauth20
- Add AuthController and GoogleStrategy
- Update TypeORM version (0.2 -> 0.3.6)

## Next

- Add UserEntity for storing user data.
- JWT Service

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ import { AppModule } from './app.module';
 
 async function bootstrap() {
   const app = await NestFactory.create(AppModule);
+  app.setGlobalPrefix('/api');
```
Comment (marshallku) [inline src/main.ts]: We don't have global prefix

**Review by marshallku — CHANGES_REQUESTED** [review]

File: src/main.ts (hunk 1 above)
Comment (0sinf) [reply src/main.ts]: Oh! I got it.

Comment (0sinf) [reply src/main.ts]: I fix it.

**Review by marshallku — APPROVED** [review]
LGTM - Gj!

**Pull request merged**